node/src/components/consensus/highway_core/highway.rs

```diff
@@ -511,8 +511,15 @@ impl<C: Context> Highway<C> {
         rng: &mut dyn CryptoRngCore,
     ) -> Vec<Effect<C>> {
         let vote_hash = swvote.hash();
+        let creator = swvote.wire_vote.creator;
         self.state.add_valid_vote(swvote);
-        self.on_new_vote(&vote_hash, now, rng)
+        let mut evidence_effects = self
+            .state
+            .opt_evidence(creator)
+            .map(|ev| vec![Effect::NewVertex(ValidVertex(Vertex::Evidence(ev.clone())))])
```

Comment:
You probably mean to only gossip the evidence if it's new, do you? This would send it to everyone whenever we add any vote by the equivocator.

Reply:
Actually, I thought it's fine to gossip every evidence b/c we shouldn't be adding any more of equivocator's votes to the state after it's marked as equivocator. Although, in the case where we detect an equivocation _while_ we're still downloading dependencies of an _back-then-honest_ vote we would gossip it unnecessarily.

Reply:
But we currently don't even filter out additional incoming votes by equivocators. We'd just add them to the state. Also, we already gossip any two votes that cause us to create an `Evidence`, so it's currently redundant.

Given the current state of the code, wouldn't it be better to just not gossip `Evidence` for now? (I though that was yesterday's conclusion?)

Reply:
Yes, but I thought it's an easy ticket and will at least improve how quickly equivocation is propagating around the network. It will also improve the spam protection story as endorsements will be created earlier. The cost isn't high either – just one `Vertex::Evidence` message sent.

I can close the PR if you insist though.